linux/Platform.c

```diff
@@ -854,3 +859,134 @@ void Platform_getBattery(double* percent, ACPresence* isOnAC) {
    Platform_Battery_cacheIsOnAC = *isOnAC;
    Platform_Battery_cacheTime = now;
 }
+
+char** Platform_getLocalIPv4addressChoices() {
+   int s = socket(AF_INET, SOCK_STREAM, 0);
+   if (s < 0)
+      return NULL;
+
+   struct ifconf ifconf;
+   char ifBuffer[4096];
+   ifconf.ifc_buf = ifBuffer;
+   ifconf.ifc_len = sizeof(ifBuffer);
+
+   if (ioctl(s, SIOCGIFCONF, &ifconf) == -1) {
+      close(s);
+      return NULL;
+   }
+
+   close(s);
+
+   const struct ifreq *ifr = (const struct ifreq *)ifBuffer;
+   unsigned int count = ifconf.ifc_len / sizeof(*ifr);
+
+   char** ret = xMalloc((count + 1) * sizeof(char**));
+
+   for (unsigned int i = 0; i < count; i++)
+      ret[i] = xStrdup(ifr[i].ifr_name);
+
+   ret[count] = NULL;
+
+   return ret;
+}
+
+void Platform_getLocalIPv4address(const char* choice, char* buffer, int size) {
+   assert(size > 0);
+
+   int s = socket(AF_INET, SOCK_STREAM, 0);
+   if (s < 0) {
+      xSnprintf(buffer, size, "N/A");
+      return;
+   }
+
+   struct ifconf ifconf;
+   char ifBuffer[4096];
+   ifconf.ifc_buf = ifBuffer;
+   ifconf.ifc_len = sizeof(ifBuffer);
+
+   if (ioctl(s, SIOCGIFCONF, &ifconf) == -1) {
+      close(s);
+      xSnprintf(buffer, size, "N/A");
+      return;
+   }
+
+   close(s);
+
+   const struct ifreq *ifr = (const struct ifreq *)ifBuffer;
+   unsigned int ifs = ifconf.ifc_len / sizeof(*ifr);
+   for (unsigned int i = 0; i < ifs; i++) {
+      if (choice && (0 != strcmp(ifr[i].ifr_name, choice)))
+         continue;
+
+      const void *ip_addr = (const void *)&(((const struct sockaddr_in *)(const void *)&(ifr[i].ifr_addr))->sin_addr);
+
+      if (!inet_ntop(AF_INET, ip_addr, buffer, size))
+         xSnprintf(buffer, size, "N/A");
+
+      return;
+   }
+
+   xSnprintf(buffer, size, "N/A");
+}
+
+char** Platform_getLocalIPv6addressChoices() {
+   FILE* file = fopen(PROCDIR "/net/if_inet6", "r");
+   if (!file)
+      return NULL;
+
+   struct in6_addr s6;
+   char ifName[16];
+   unsigned int count = 0;
+   char** ret = xMalloc(sizeof(char**));
```

Comment:
`sizeof`'d type mismatches intended type:
```suggestion
   char** ret = xMalloc(sizeof(char*));
```